sbom-utility-scripts/scripts/base-images-sbom-script/app/base_images_sbom_script.py

```diff
@@ -127,10 +136,53 @@ def main():
         sbom = json.load(f)
 
     base_images_sbom_components = get_base_images_sbom_components(base_images_digests, is_last_from_scratch)
-    if "formulation" in sbom:
-        sbom["formulation"].append({"components": base_images_sbom_components})
+    if args.sbom_type == "cyclonedx":
+        if "formulation" in sbom:
+            sbom["formulation"].append({"components": base_images_sbom_components})
+        else:
+            sbom.update({"formulation": [{"components": base_images_sbom_components}]})
     else:
-        sbom.update({"formulation": [{"components": base_images_sbom_components}]})
+        packages = []
+        relationships = []
+        for component in base_images_sbom_components:
+            SPDXID = (
+                f"SPDXRef-{component['type']}-{component['name']}-"
+                + f"{hashlib.sha256(component['purl'].encode()).hexdigest()}"
+            )
+            packages.append(
+                {
+                    "SPDXID": SPDXID,
+                    "name": component["name"],
+                    "externalRefs": [
+                        {
+                            "referenceCategory": "PACKAGE-MANAGER",
+                            "referenceType": "purl",
+                            "referenceLocator": component["purl"],
+                        }
+                    ],
+                    "annotations": [
+                        {
+                            "annotator": "konflux",
+                            "annotationDate": datetime.datetime.now().isoformat(),
+                            "annotationType": "OTHER",
+                            "comment": json.dumps(
+                                {"name": property["name"], "value": property["value"]},
+                                separators=(",", ":"),
+                            ),
+                        }
```

Comment:
What is the intended interface here? Are we stating that consumers of Konflux-made SPDX SBOMs should expect all the `annotations` with `annotator=konflux` to be in the form `{"name": "...", "value":  "..."}`?

Reply:
Cachi2 uses some cyclonedx properties, however spdx format (v2.3) doesn't have any field which we could use for that, so we decided to use package annotations for that.

Reply:
Right, I remember now. The format looks weird, but I don't really have a better idea. It will just need to be documented somewhere.

Reply:
I guess cachi2 would be the best place to document this.

Reply:
For the cachi2 annotations, I would agree. I guess those use `"annotator": "cachi2"`?

For the Konflux annotations, probably somewhere in https://github.com/konflux-ci/docs

Reply:
We now have and ADR that mentions the annotations, should be good enough